openerp/addons/base/tests/test_qweb.py

```diff
@@ -85,6 +86,12 @@ def __call__(self, name, options):
                 arch = etree.tostring(root, encoding='utf-8', xml_declaration=True)
                 return arch
 
+def stringify_children(node):
+    parts = ([node.text] +
+            list(chain(*([c.text, etree.tostring(c), c.tail] for c in node.getchildren() if not isinstance(c, etree._Comment)))) +
+            [node.tail])
+    # filter removes possible Nones in texts and tails
+    return ''.join(filter(None, parts))
```

Comment:
line 90 creates a list, line 91 unpacks a generator into a possibly huge tuple (of lists) and creates a list, line 92 creates a list, lines 90~92 then create a new list from the previous 3, and line 94 create a new (filtered) list from everything else.

All of the list creation except the inner one (of text items from non-comment nodes) should be avoidable.